### PR TITLE
Add zsh completion script

### PR DESCRIPTION
**Pull request by Geod24**
Currently supports the main dub commands: build, init, run, test, clean, dustmite...
Reads the local packages from dub whenever possible.
A few improvements to dub will be necessary to make it completely useful (e.g. display configurations).

### CODE REVIEW

**Comment by Geod24** [comment]
Guess we don't have that many people that can review a zsh completion script. Self merging because there's no way this breaks anything.

**Comment by PetarKirov** [comment]
@Geod24 Can we generate shell completion scripts automatically? AFAIR we already generate CLI help, DDoc for dlang.org and man pages out of dmd's CLI.

**Comment by Geod24** [comment]
For posterity (discussed on Slack): I would love to, but it seemed like a faster way to just fo with a manual script at first. Also the real value will be in completing things based on the user / project settings (e.g. completing `-c` / `--config`), for which we first need to add support to dub.

TL;DR: Ideally yes, but not soon (from me).

**Comment by wilzbach** [comment]
We already generate the man pages for dub automatically - https://github.com/dlang/dub/blob/master/scripts/man/gen_man.d.

This might help.
